pkg/engine/diff.go

```diff
@@ -436,7 +448,9 @@ func printOldNewDiffs(
 	if diff := olds.Diff(news); diff != nil {
 		printObjectDiff(b, *diff, planning, indent, summary, debug)
 	} else {
-		printObject(b, news, planning, indent, op, true, debug)
+		// If there's no diff, report the op as Same - there's no diff to render
+		// so it should be rendered as if nothing changed.
+		printObject(b, news, planning, indent, deploy.OpSame, true, debug)
```

Comment:
I'm thinking it's potentially the case that `printObject` doesn't need a `StepOp` parameter anymore, but I didn't want to go too crazy refactoring this whole file for a bug fix. I'm happy to try to get rid of it if anyone wants me to.

Reply:
anything to simplify would be good :)  just make separate commits and i'm happy.

Reply:
Ah, actually I was wrong - we do still need a `StepOp` here in case we are printing an object that got created.